app/routes.py

```diff
@@ -1,2 +1,245 @@
+            return jsonify({"details" : "Invalid data"}), 400
+
+        new_task = Task(title=title, description=description, completed_at=completed_at)
+
+        if new_task.completed_at:
+            new_task.is_complete = True
+
+        db.session.add(new_task)
+        db.session.commit()
+
+        return jsonify({"task" : {
+            "id" : new_task.task_id,
+            "title" : new_task.title,
+            "description" : new_task.description,
+            "is_complete" : new_task.is_complete
+        }}), 201
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+    
+    if request.method == "GET":
+        if task.goal_id:
+            return {"task" : {
+            "id" : task.task_id,
+            "goal_id" : task.goal_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+        }}
+        return {"task" : {
+            "id" : task.task_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+        }}
+    elif request.method == "PUT":
+        request_body = request.get_json()
+
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+
+        if request_body.get("completed_at"):
+            task.is_complete = True
+
+        db.session.commit()
+
+        return jsonify({"task" : {
+            "id" : task.task_id,
+            "title" : task.title,
+            "description" : task.description,
+            "is_complete" : task.is_complete
+        }}), 200
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return jsonify({"details" : f'Task {task.task_id} "{task.title}" successfully deleted'})
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_tasks_complete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    task.is_complete = True
+    task.completed_at = datetime.utcnow()
+
+    return jsonify({ "task" : {
+        "id" : task.task_id,
+        "title" : task.title,
+        "description" : task.description,
+        "is_complete" : task.is_complete
+    }}), 200
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_tasks_incomplete(task_id):
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return jsonify(None), 404
+
+    # request_body = request.get_json()
+
+    task.is_complete = False
+    task.completed_at = None
+
+    return jsonify({ "task" : {
+        "id" : task.task_id,
+        "title" : task.title,
+        "description" : task.description,
+        "is_complete" : task.is_complete
+    }}), 200
+
+@goals_bp.route("", methods=["GET","POST"])
+def handle_goals():
```

Comment:
Think about which comments about the task routes could also apply to the goal routes.